### PR TITLE
Set highest pkg version+pluginDownloadURL on `pulumi convert`

### PR DESCRIPTION
**Pull request by aq17**
Extends https://github.com/pulumi/pulumi-yaml/pull/331 by setting `pluginDownloadURL` as well on resources

Todo: invokes

### CODE REVIEW

File: pkg/pulumiyaml/codegen/load.go [hunk 1]
```diff
@@ -712,10 +725,14 @@ func (imp *importer) importResource(kvp ast.ResourcesMapEntry, highestPkgVersion
...
-	if resource.Options.Version != nil {
+	if p, ok := highestPkgInfo[pkg.Name()]; ok {
 		resourceOptions.Body.Items = append(resourceOptions.Body.Items, &model.Attribute{
 			Name:  "version",
-			Value: quotedLit(highestPkgVersion[pkg.Name()]),
+			Value: quotedLit(p.version),
+		})
+		resourceOptions.Body.Items = append(resourceOptions.Body.Items, &model.Attribute{
+			Name:  "pluginDownloadURL",
+			Value: quotedLit(p.pluginDownloadURL),
```
Comment (iwahbe) [inline pkg/pulumiyaml/codegen/load.go:735]: We should probably check that `p.pluginDownloadURL != ""`. Same for `p.version`

File: pkg/tests/transpiled_examples/webserver-json/program.pp [hunk 2]
```diff
@@ -1,51 +1,57 @@
...
+        version = "4.37.1"
+        pluginDownloadURL = ""
+    }
 }
 
 output instanceId {
-	__logicalName = "InstanceId"
-	value = webServer.id
+    __logicalName = "InstanceId"
+    value = webServer.id
```
Comment (iwahbe) [inline pkg/tests/transpiled_examples/webserver-json/program.pp:46]: Why did this happen?

**Review by iwahbe — APPROVED** [review]
The change looks good, pending 🟢 tests.

**Review by AaronFriel — COMMENTED** [review]
Could you update our examples with a test that uses a pluginDownloadUrl? You might need to ask in Slack for a good example of a third party provider we would want to test.

**Comment by aq17** [comment]
> Could you update our examples with a test that uses a pluginDownloadUrl? You might need to ask in Slack for a good example of a third party provider we would want to test.

Sounds good!

**Pull request merged**